packages/@coorpacademy-progression-engine/src/compute-next-step/rule-engine/apply-instructions.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+import assign from 'lodash/fp/assign';
+import reduce from 'lodash/fp/reduce';
+import set from 'lodash/fp/set';
+import update from 'lodash/fp/update';
+import type {State} from '../../types';
+import type {Instruction, Variables} from './types';
+
+const apply = (target: Variables, instruction: Instruction): Variables => {
+  const {type, value, field} = instruction;
+  switch (type) {
+    case 'set':
+      return set(field, value, target);
+    case 'add':
+      return update(field, v => v + value, target);
+    default:
+      return target;
```

Comment:
`error` Not yet Implemented?

Reply:
ou pas de `default` (pas testable)

Reply:
@esa-coorp oui, c'est testable, on met juste une instruction avec un type : 'foo'